device_backends/uio/src/UioBackend.cc

```diff
@@ -108,14 +96,14 @@ namespace ChimeraTK {
         numberOfInterrupts = _uioAccess->waitForInterrupt(100);
       }
       catch(ChimeraTK::runtime_error& ex) {
-        setException();
+        setException(ex.what());
         continue;
       }
 
       if(numberOfInterrupts > 0) {
         _uioAccess->clearInterrupts();
 
-        if(_hasActiveException) {
+        if(isFunctional()) {
```

Comment:
There is a NOT missing:
```suggestion
        if(!isFunctional()) {
```